fetcher/mempool.go

```diff
@@ -28,20 +28,27 @@ import (
 func (f *Fetcher) Mempool(
 	ctx context.Context,
 	network *types.NetworkIdentifier,
-) ([]*types.TransactionIdentifier, error) {
-	response, _, err := f.rosettaClient.MempoolAPI.Mempool(
+) ([]*types.TransactionIdentifier, *Error) {
+	response, clientErr, err := f.rosettaClient.MempoolAPI.Mempool(
 		ctx,
 		&types.NetworkRequest{
 			NetworkIdentifier: network,
 		},
 	)
 	if err != nil {
-		return nil, fmt.Errorf("%w: /mempool %s", ErrRequestFailed, err.Error())
+		res := &Error{
```

Comment:
thoughts about renaming `res` to something like `fetcherError`. This is not really a response I think?

Reply:
makes sense, fixed :)